File: golang/partitioning-into-minimum-number-of-deci-binary-numbers/partitioning-into-minimum-number-of-deci-binary-numbers.go
package main

import (
	"fmt"
)

func minPartitions(n string) int {
	maxDigit := 0

	for _, char := range n {
		digit := int(char - '0')

		if digit > maxDigit {
			maxDigit = digit
		}
	}

	return maxDigit
}

func runTest(caseNum int, n string, expected int) {
	result := minPartitions(n)
	status := "FAIL"
	if result == expected {
		status = "PASS"
	}

	fmt.Printf("Test Case %d:\n", caseNum)
	fmt.Printf("  Input:    %s\n", n)
	fmt.Printf("  Output:   %d\n", result)
	fmt.Printf("  Expected: %d\n", expected)
	fmt.Printf("  Status:   %s\n\n", status)
}

func main() {
	runTest(1, "32", 3)
	runTest(2, "82734", 8)
	runTest(3, "27346209830709182346", 9)
	runTest(4, "55555", 5)
	runTest(5, "1", 1)
}

File: leetcode/partitioning-into-minimum-number-of-deci-binary-numbers.py
class Solution(object):
    def minPartitions(self, n):
        return int(max(n))


if __name__ == "__main__":
    solver = Solution()

    def run_test(case_num, n, expected):
        result = solver.minPartitions(n)
        status = "PASS" if result == expected else "FAIL"

        print(f"Test Case {case_num}:")
        print(f"  Input:    {n}")
        print(f"  Output:   {result}")
        print(f"  Expected: {expected}")
        print(f"  Status:   {status}\n")

    # Example 1
    # Max digit in "32" is 3.
    run_test(1, "32", 3)

    # Example 2
    # Max digit in "82734" is 8.
    run_test(2, "82734", 8)

    # Example 3
    # Max digit is 9.
    run_test(3, "27346209830709182346", 9)

    # Custom Case: All same digits
    run_test(4, "55555", 5)

    # Custom Case: Single digit
    run_test(5, "1", 1)
